File: api/Params/Admin/VisaParams.cs
namespace api.Params.Admin
{
    public class VisaParams: PaginationParams
    {
        public int Id { get; set; }
        public int VisaItemId { get; set; }
        public string VisaNo { get; set; }
        public int CvRefId { get; set; }
        public int CustomerId { get; set; }
        public string CustomerName { get; set; }
        public string VisaExpiryH { get; set; }
        public DateTime VisaExpiryG { get; set; }
        public string VisaSponsorName { get; set; }
        public string VisaCategory { get; set; }
        public string VisaConsulate { get; set; }
        public bool VisaCanceled { get; set; }
        public int DepItemId { get; set; }
        public bool VisaApproved { get; set; }
    }
}

File: client/src/app/_models/params/Admin/visaParams.ts
export class visaParams
{
    id : number=0;
    visaItemId: number = 0;
    visaNo: string = '';
    customerId: number = 0;
    customerName: string = '';
    cvRefId: number = 0;
    visaExpiryH: string = '';
    visaExpiryG: Date = new Date(1900,1,1);
    visaSponsorName: string = '';
    
    visaCategory: string='';
    visaConsulate: string='';
    visaCanceled: boolean = false;

    depItemId: number = 0;
    visaApproved: boolean = false;

    sort = "visano";
    pageNumber = 1;
    pageSize = 15;
    search: string='';
}